### PR TITLE
Extract comparison logic

### PR DESCRIPTION
**Pull request by IanWhitney**
- Variable name improvements
- Extracting the logic that did the comparisons into one class, more details on why in 042af80

### CODE REVIEW

File: lib/query_string_search/comparator.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    def initialize(subject)
+      self.subject = subject
+    end
+
+    def equal?(other)
+      normalize(subject) == normalize(other)
+    end
+
+    def normalize(unnormalized)
```
Comment (IanWhitney) [inline lib/query_string_search/comparator.rb:17]: This type-checking if will be going away in a later refactoring.

File: lib/query_string_search/abstract_matcher.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module QueryStringSearch
   class AbstractMatcher
-    attr_accessor :attribute, :value
+    attr_accessor :attribute, :desired_value
```
Comment (DebbieGillespie) [inline lib/query_string_search/abstract_matcher.rb:3]: Is attribute the best name for the attribute?  It is somewhat confusing.  I don't have any suggestions so attribute may be the best option.

**Comment by DebbieGillespie** [comment]
:+1:

**Comment by IanWhitney** [comment]
This closes #6

**Pull request merged**